components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -504,46 +490,28 @@ void CosmeticFiltersJSHandler::ApplyRules(bool de_amp_enabled) {
 
   CSSRulesRoutine(*resources_dict_);
 
-  bool has_removals = false;
-  //: remove()
-  std::string remove_selectors_json;
-  const auto* remove_selectors_list =
-      resources_dict_->FindList("remove_selectors");
-  if (remove_selectors_list && !remove_selectors_list->empty()) {
-    base::JSONWriter::Write(*remove_selectors_list, &remove_selectors_json);
-    has_removals = true;
-  } else {
-    remove_selectors_json = "undefined";
-  }
-
-  //: remove_classes
-  std::string remove_classes_json;
-  const auto* remove_classes_dictionary =
-      resources_dict_->FindDict("remove_classes");
-  if (remove_classes_dictionary && !remove_classes_dictionary->empty()) {
-    base::JSONWriter::Write(*remove_classes_dictionary, &remove_classes_json);
-    has_removals = true;
-  }
-
-  //: remove_attrs
-  std::string remove_attrs_json;
-  const auto* remove_attrs_dictionary =
-      resources_dict_->FindDict("remove_attrs");
-  if (remove_attrs_dictionary && !remove_attrs_dictionary->empty()) {
-    base::JSONWriter::Write(*remove_attrs_dictionary, &remove_attrs_json);
-    has_removals = true;
-  }
-
-  if (has_removals) {
-    // Building a script for removals
-    std::string new_selectors_script = base::StringPrintf(
-        kRemovalsInjectScript, remove_selectors_json.c_str(),
-        remove_classes_json.c_str(), remove_attrs_json.c_str());
-    web_frame->ExecuteScriptInIsolatedWorld(
-        isolated_world_id_,
-        blink::WebScriptSource(
-            blink::WebString::FromUTF8(new_selectors_script)),
-        blink::BackForwardCacheAware::kAllow);
+  const std::string* procedural_actions_script =
+      resources_dict_->FindString("procedural_actions_script");
+  if (procedural_actions_script) {
+    v8::Isolate* isolate = web_frame->GetAgentGroupScheduler()->Isolate();
+    v8::HandleScope handle_scope(isolate);
+
+    v8::Local<v8::Value> v8_stylesheet =
+        web_frame->ExecuteScriptInIsolatedWorldAndReturnValue(
+            isolated_world_id_,
+            blink::WebScriptSource(
+                blink::WebString::FromUTF8(*procedural_actions_script)),
+            blink::BackForwardCacheAware::kAllow);
+
+    CHECK(v8_stylesheet->IsString());
```

Comment:
This seems dangerous as some script error might result in this being something other than a string or it could return nothing. I think it's better to do something like `if (!v8_stylesheet.IsEmpty() && v8_stylesheet->String())`

Reply:
- [x] updated